napari/_vispy/visuals/volume.py

```diff
@@ -36,75 +36,9 @@
     adjacent_bg = adjacent_bg * int( floatEqual(val_pos, categorical_bg_value) );
     return adjacent_bg;
 }
+"""
 
-vec3 calculateGradient(vec3 loc, vec3 step, out int n_bg_borders) {
-    // calculate gradient within the volume by finite differences
-
-    n_bg_borders = 0;
-    vec3 G = vec3(0.0);
-
-    float prev;
-    float next;
-
-    prev = colorToVal($get_data(loc - vec3(step.x, 0, 0)));
-    next = colorToVal($get_data(loc + vec3(step.x, 0, 0)));
-    n_bg_borders += detectAdjacentBackground(prev, next);
-    G.x = next - prev;
-
-    prev = colorToVal($get_data(loc - vec3(0, step.y, 0)));
-    next = colorToVal($get_data(loc + vec3(0, step.y, 0)));
-    n_bg_borders += detectAdjacentBackground(prev, next);
-    G.y = next - prev;
-
-    prev = colorToVal($get_data(loc - vec3(0, 0, step.z)));
-    next = colorToVal($get_data(loc + vec3(0, 0, step.z)));
-    n_bg_borders += detectAdjacentBackground(prev, next);
-    G.z = next - prev;
-
-    return G;
-}
```

Comment:
Re: https://github.com/napari/napari/pull/7431#discussion_r1872254529
This is the previous "fast" method being removed.